api/resource.yaml

```diff
@@ -0,0 +1,2200 @@
+
+servers:
+  - url: https://{host}:{port}
+    variables:
+      host:
+        default: "localhost"
+      port:
+        default: "8090"
+
+tags:
+  - name: resource-servers
+    description: Operations related to resource server management
+  - name: resources
+    description: Operations related to resource management within resource servers
+  - name: actions
+    description: Operations related to action management at resource server and resource levels
+  - name: permissions
+    description: Operations to preview and retrieve derived permissions
+
+security:
+  - OAuth2: [system]
+
+paths:
+  /resource-servers:
+    get:
+      tags:
+        - resource-servers
+      summary: List resource servers
+      parameters:
+        - $ref: '#/components/parameters/limitQueryParam'
+        - $ref: '#/components/parameters/offsetQueryParam'
+      responses:
+        "200":
+          description: List of resource servers
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResourceServerListResponse'
+              example:
+                totalResults: 5
+                startIndex: 1
+                count: 2
+                resourceServers:
+                  - id: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                    name: "booking"
+                    description: "Handles all booking operations"
+                    identifier: "booking-system"
+                    ouId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+                  - id: "257e528f-eb24-48b6-884d-20460e190957"
+                    name: "property"
+                    description: "Property and facility management"
+                    identifier: "property-mgmt"
+                    ouId: "a839f4bd-39dc-4eaa-b5cc-210d8ecaee87"
+                links:
+                  - href: "resource-servers?offset=2&limit=2"
+                    rel: "next"
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                invalid-limit:
+                  summary: Invalid limit parameter
+                  value:
+                    code: "RES-1011"
+                    message: "Invalid limit parameter"
+                    description: "The limit parameter must be a positive integer"
+                invalid-offset:
+                  summary: Invalid offset parameter
+                  value:
+                    code: "RES-1012"
+                    message: "Invalid offset parameter"
+                    description: "The offset parameter must be a non-negative integer"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-5000"
```

Comment:
Let's use the new server level InternalServerError struct instead of package specific error codes.
We should eventually get rid of other existing package level server errors as well.

Check other places and the implementation as well.

Ref: https://github.com/asgardeo/thunder/blob/4ed39980c0ed3efb93ee56cb51efa69da9927920/backend/internal/system/error/serviceerror/error.go#L57